src/intrinsic.rs

```diff
@@ -240,6 +240,28 @@ pub trait EvalContextExt<'a, 'mir, 'tcx: 'a+'mir>: crate::MiriEvalContextExt<'a,
                 this.binop_ignore_overflow(op, a, b, dest)?;
             }
 
+            "minnumf32" | "maxnumf32" => {
+                let a = this.read_scalar(args[0])?.to_f32()?;
+                let b = this.read_scalar(args[1])?.to_f32()?;
+                let res = if intrinsic_name.get().starts_with("min") {
```

Comment:
For this we can use libapfloat, in fact, you can just not do the `to_f32` part and call `this.binary_op(mir::BinOp::Lt, a.into(), b.into())?.to_bool()?` and decide that way whether to write `a` or `b` into `dest`. This way the code paths for `f64` and `f32` are unified.

Reply:
But that way I have to implement all the crazy NaN stuff myself. :/

Implementing min/max [is complicated](https://medium.com/@nettsundere/the-implementation-of-min-and-max-operations-for-floating-point-types-80d3616c084d) and the naive one you suggested is likely wrong.

Reply:
Cc @eddyb for floating point knowledge ;)

Reply:
well, apfloat has `min` and `max`, so you can `from_bits` for `Double` or `Single` depending on the size and then call `min` or `max` respectively. https://github.com/rust-lang/rust/blob/5c45343f11fbf93cf4e15568aee3ff3f2f287466/src/librustc_apfloat/lib.rs#L434

won't remove the duplication, but we'll be host independent

Reply:
Ah, good point.

I should probably add `to`/`from`-apfloat methods to `Scalar`. And I am wondering if I should remove the to/from-f32 ones? Seems like a footgun to accidentally use host FP.

Reply:
Your suggestion indeed fails one of those tests btw, still figuring out which so I can hoist at least that one to the Miri test suite.

Reply:
I'm confused. Did you try to use the `min` or `max` method? If these are doing something wrong, that's a bug in them, but last I looked they had some special logic to handle `NaN`

Reply:
it looks to me like the runtime code and the libapfloat code do the same?

https://doc.rust-lang.org/src/core/num/f64.rs.html#411-421

vs 

https://github.com/rust-lang/rust/blob/5c45343f11fbf93cf4e15568aee3ff3f2f287466/src/librustc_apfloat/lib.rs#L434

Reply:
I *am* using apfloat's `min` and `max` methods...?